### PR TITLE
Add "sample-public-image-"prefixed tags to images

### PR DESCRIPTION
**Pull request by NimJay**
## Description

* Please see comment in Google-internal bug b/370252964#comment2.

## Tasks

* [ ] Merge this pull-request for me once it is approved.

### CODE REVIEW

**Comment by iennae** [comment]
we can use b links but guidance is not make an active link

**Review by iennae — APPROVED** [review]
LGTM; thanks for doing this critical work!

**Comment by NimJay** [comment]
Thanks for much for reviewing this, Jennifer!

We'll likely close this pull-request in favour of b/374760962. But I'll wait until b/374760962 is resolved.
Looks like there's a merge conflict I'd have to resolve anyway.

**Comment by iennae** [comment]
This won't work as the images need to be unique, hence the use of the COMMIT_SHA to help make the tag unique

File: batch/aiml-workloads/src/cloudbuild.yaml [hunk 1]
```diff
@@ -22,8 +22,11 @@ steps:
...
     - '-t'
     - 'us-docker.pkg.dev/google-samples/containers/gke/batch-ml-workload'
+    - '-t'
+    - 'us-docker.pkg.dev/google-samples/containers/gke/batch-ml-workload:sample-public-image-${SHORT_SHA}'
     - '.'
   dir: 'batch/aiml-workloads/src'
 
 images:
   - 'us-docker.pkg.dev/google-samples/containers/gke/batch-ml-workload'
+  - 'us-docker.pkg.dev/google-samples/containers/gke/batch-ml-workload:sample-public-image-${SHORT_SHA}'
```
Comment (NimJay) [inline batch/aiml-workloads/src/cloudbuild.yaml:32]: `$SHORT_SHA` = the first seven characters of COMMIT_SHA
See [Cloud Build docs here](https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values#using_default_substitutions).

Comment (iennae) [reply batch/aiml-workloads/src/cloudbuild.yaml:32]: awesome that works :D

**Review by iennae — APPROVED** [review]
Fantastic! THANK YOU!

**Pull request merged**